notebook/static/notebook/js/menubar.js

```diff
@@ -128,6 +128,142 @@ define([
 
         this._new_window(url);
     };
+    
+    MenuBar.prototype._nbconvert_upload_conf = function (download) {
+      
+      var body =  $('<div>');
+      var notebook_path = utils.encode_uri_components(this.notebook.notebook_path);
+
+      var create_json = function(notebook, config){
+        var json_to_pass = {
+          notebook: notebook.toJSON(),
+          config: config
+        };
+        return json_to_pass;
```

Comment:
I can change the API more fundamentally, but then the name should probably be passed in explicitly as part of the JSON object.